### ISSUE TITLE
Fasta accepts conflicting interface type arguments when contravariant

### ISSUE THREAD

**Issue by askeksa-google** [dart-lang/sdk#34574]
```dart
class I<T> {}

class A {}

class B extends A {}

class X implements I<B> {}

class Y extends X implements I<A> {}
```
Is invalid, since `Y` implements two different generic instantiations of `I`. But Fasta does not report an error. If `A` and `B` are swapped, an error is correctly reported.

**Comment by kmillikin** [comment]
This'll be fixed by https://dart-review.googlesource.com/c/sdk/+/76141

As far as I can tell, the code that allows this case was to deal with VM-style super mixins, and probably before mixin type inference was implemented.

**Comment by kmillikin** [comment]
Right, it was this P0 issue affecting Flutter: https://github.com/dart-lang/sdk/issues/31656

The special case was a "short-term hack" https://dart-review.googlesource.com/c/sdk/+/33884

**Comment by kmillikin** [comment]
Fixed by c9b05d3f1ff0ba22cc8cc2e66e2654d99609e087
